ipatests/test_integration/test_installation.py

```diff
@@ -772,3 +773,49 @@ def test_files_ownership_and_permission_teardown(self):
         """ Method to restore the default bashrc contents"""
         if self.bashrc_file is not None:
             self.master.put_file_contents('/root/.bashrc', self.bashrc_file)
+
+
+class TestInstallMasterReplica(IntegrationTest):
+    """https://pagure.io/freeipa/issue/7929
+    """
+    num_replicas = 1
+    topology = 'star'
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_kra=True)
+        # do not install KRA on replica, it is part of test
+        tasks.install_replica(cls.master, cls.replicas[0], setup_kra=False)
+
+    def test_replicamanage_del(self):
+        """Test Steps:
+        1. Setup server
+        2. Setup replica
+        3. modify the replica entry on Master:
+           ldapmodify -D cn="Directory Manager"-w <passwd>
+           dn: cn=KDC,cn=<replicaFQDN>,cn=masters,cn=ipa,cn=etc,<baseDN>
+           changetype: modify
+           replace: ipaconfigstring
+           ipaconfigstring: pkinitEnabled
+           ipaconfigstring: configuredService
+        4. On master,
+           run ipa-replica-manage del <replicaFQDN> --cleanup --force
+        """
+        # https://pagure.io/freeipa/issue/7929
+        # modify the replica entry on Master
+        cmd_output = None
+        dn_entry = 'dn: cn=KDC,cn=%s,cn=masters,cn=ipa,' \
+                   'cn=etc,dc=testrelm, dc=test' % self.replicas[0].hostname
```

Comment:
This has the side-effect of dropping ipaConfigString: startOrder 10 and ipaConfigString: kdcProxyEnabled.
It would be more complex, but more correct do the ldapmod as:
```
changetype: modify
delete: ipaconfigstring
ipaconfigstring: enabledService
-
add: ipaconfigstring
ipaconfigstring: configuredService
```

Reply:
made appropriate changes